library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/sync/ModProtocols.java

```diff
@@ -0,0 +1,155 @@
+ * to connect:
+ * <pre> {@code
+ * {
+ *     "quilt_loader": {...},
+ *     "quilt_registry": {
+ *         "mod_protocol": {
+ *             "value": 1,
+ *             "optional": true
+ *         }
+ *     }
+ * }
+ * }</pre>
+ * <p>
+ * In addition to mods defining their own mod protocols, users can define a special modpack protocol in the Quilt Registry
+ * config (located at {@code configs/quilt/qsl/registry.toml}). Other than being defined by the user, the modpack protocol
+ * is otherwise identical to standard mod protocols.
+ */
+@ApiStatus.Experimental
+public final class ModProtocols {
+	public static final int UNSUPPORTED = -1;
+	private ModProtocols() {}
+
+	/**
+	 * {@return {@code true} if a modpack protocol has been provided through the config, otherwise {@code false}}
+	 */
+	@Contract(pure = true)
+	public static boolean isModpackProtocolEnabled() {
+		return ModProtocolImpl.modpackDef != null;
+	}
+
+	/**
+	 * {@return a {@code String} representing the modpack protocol id, otherwise {@code null}}
+	 */
+	@Contract(pure = true)
+	@Nullable
+	public static String getModpackProtocolId() {
+		return isModpackProtocolEnabled() ? ModProtocolImpl.modpackDef.id() : null;
+	}
+
+	/**
+	 * {@return a {@code String} representing the modpack protocol display name, otherwise {@code null}}
+	 */
+	@Contract(pure = true)
+	@Nullable
+	public static String getModpackDisplayName() {
+		return isModpackProtocolEnabled() ? ModProtocolImpl.modpackDef.displayName() : null;
+	}
+
+	/**
+	 * Checks the mod protocol version supported by a given player.
+	 *
+	 * @param player Player to check against
+	 * @param modContainer ModContainer defining protocol version
+	 * @return latest supported by player protocol for mod. -1 if not supported
+	 */
+	@Contract(pure = true)
+	public static int getSupported(@NotNull ServerPlayerEntity player, @NotNull ModContainer modContainer) {
+		return player.networkHandler != null ? getSupported(player.networkHandler, modContainer) : UNSUPPORTED;
+	}
+
+	/**
+	 * Checks the mod protocol version supported by a given player.
+	 *
+	 * @param handler player's network handler to check against
+	 * @param modContainer ModContainer defining protocol version
+	 * @return latest supported by player protocol for mod. -1 if not supported
+	 */
+	@Contract(pure = true)
+	public static int getSupported(@NotNull ServerPlayNetworkHandler handler, @NotNull ModContainer modContainer) {
+		return ExtendedConnectionClient.from(handler).quilt$getModProtocol("mod:" + modContainer.metadata().id());
+	}
+
+	/**
+	 * Checks the modpack protocol version supported by a given player. Returns -1 if the server and player's
+	 * {@linkplain #getModpackProtocolId() modpack protocol id} do not match.
+	 * @param handler player's network handler to check against
+	 * @return latest supported by player protocol for this server's {@linkplain #getModpackProtocolId() modpack protocol id}. -1 if not supported
```

Comment:
```suggestion
	 * Checks the modpack protocol version supported by a given player. Returns -1 if the server and player's
	 * {@linkplain #getModpackProtocolId() modpack protocol id} do not match.
	 *
	 * @param player the player to check against
	 * @return -1 if not supported, or the latest modpack protocol version supported by the player otherwise.
```
hm, there is a bad copy-and-paste going on here